### PR TITLE
Supported a websocket in node api

### PR DESCRIPTION
**Pull request by sirano11**
## Proposed changes

This PR supported web socket protocol in node api. 
Also, It applied a latest caver-java version.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/ground-x/caver-java-ext-kas/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/ground-x/caver-java-ext-kas)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java [hunk 1]
```diff
@@ -103,57 +112,113 @@ public void initKASAPI(String chainId, String accessKeyId, String secretAccessKe
...
+     * caver.initNodeAPI(1001, "accessKey", "secretAccessKey", true); // use Websocket provider
+     * }
+     * </pre>
+     *
+     * @param chainId The Klaytn network chain id.
+     * @param accessKeyId The access key provided by KAS console.
+     * @param secretAccessKey The secret key provided by KAS console.
+     * @param isWebSocketUsed If true, HttpProvider is used. If false, WebsocketProvider is used.
+     */
+    public void initNodeAPI(int chainId, String accessKeyId, String secretAccessKey, boolean isWebSocketUsed) {
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java:142]:
```suggestion
    public void initNodeAPI(int chainId, String accessKeyId, String secretAccessKey, boolean useHttp) {
```
The variable name and description is not matched. Can you change the parameter name?

File: src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java [hunk 2]
```diff
@@ -103,57 +112,113 @@ public void initKASAPI(String chainId, String accessKeyId, String secretAccessKe
...
+    /**
+     * Initialize Node API with Http provider.
      * It sets a url to default endpoint automatically.
      * @param chainId The Klaytn network chain id.
      * @param accessKeyId The access key provided by KAS console.
      * @param secretAccessKey The secret key provided by KAS console.
      */
     public void initNodeAPI(String chainId, String accessKeyId, String secretAccessKey) {
-        initNodeAPI(chainId, accessKeyId, secretAccessKey, URL_NODE_API);
+        initNodeAPI(chainId, accessKeyId, secretAccessKey, false);
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java:154]:
```suggestion
        initNodeAPI(chainId, accessKeyId, secretAccessKey, URL_NODE_API , false);
```

File: src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java [hunk 3]
```diff
@@ -103,57 +112,113 @@ public void initKASAPI(String chainId, String accessKeyId, String secretAccessKe
...
+
+    /**
+     * Initialize Node API.<br>
+     * It sets a url to default endpoint automatically.<br>
+     * It can choose provider that Http or Websocket via isWebsocketUsed parameter.
+     *
+     * <pre>
+     * {@code
+     * caver.initNodeAPI("1001", "accessKey", "secretAccessKey", false); // use Http provider
+     * caver.initNodeAPI("1001", "accessKey", "secretAccessKey", true); // use Websocket provider
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java:165]:
```suggestion
     * caver.initNodeAPI("1001", "accessKey", "secretAccessKey", false); // use Http provider
     * caver.initNodeAPI("1001", "accessKey", "secretAccessKey", true); // use Websocket provider
```
Here also please modify together with parameter name.

File: src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java [hunk 4]
```diff
@@ -103,57 +112,113 @@ public void initKASAPI(String chainId, String accessKeyId, String secretAccessKe
...
-     * Initialize Node API.
+     * Initialize Node API with Http provider.<br>
      * It sets a url to default endpoint automatically.
      * @param chainId The Klaytn network chain id.
      * @param accessKeyId The access key provided by KAS console.
      * @param secretAccessKey The secret key provided by KAS console.
      */
     public void initNodeAPI(int chainId, String accessKeyId, String secretAccessKey) {
-        initNodeAPI(chainId, accessKeyId, secretAccessKey, URL_NODE_API);
+        initNodeAPI(chainId, accessKeyId, secretAccessKey,false);
```
Comment (jimni1222) [inline src/main/java/xyz/groundx/caver_ext_kas/CaverExtKAS.java:122]: Please modify to use `true` as a default value.

**Review by jimni1222 — DISMISSED** [review]
LGTM =) Thank you for changing!

**Review by jimni1222 — DISMISSED** [review]

**Review by aeharvlee — DISMISSED** [review]

**Review by jimni1222 — APPROVED** [review]

**Pull request merged**